### PR TITLE
Option to disable save article hits

### PR DESCRIPTION
**Pull request by jurihahn**
Pull Request for Issue #22406 .

### Summary of Changes
Option for Disable Hits counter increment

### Testing Instructions
1. Open an article and check if the hits do increase
2. Go to content options and Set "Save Hits" to "NO"
3. Open an article and check if the hits do not increase

### Expected result
If disabled, article hits do not increase

### Actual result
article hits do not increase always

### Documentation Changes Required
no?

### CODE REVIEW

File: administrator/language/en-GB/en-GB.com_content.ini [hunk 1]
```diff
@@ -90,6 +90,8 @@ COM_CONTENT_FIELD_PUBLISH_DOWN_DESC="An optional date to Finish Publishing the a
 COM_CONTENT_FIELD_PUBLISH_DOWN_LABEL="Finish Publishing"
 COM_CONTENT_FIELD_PUBLISH_UP_DESC="An optional date to Start Publishing the article."
 COM_CONTENT_FIELD_PUBLISH_UP_LABEL="Start Publishing"
+COM_CONTENT_FIELD_SAVEHITS_DESC="If you want to display the hits, then select YES to save the number of hits."
+COM_CONTENT_FIELD_SAVEHITS_LABEL="Save Hits"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_content.ini]: ==>
"Record Hits"

File: administrator/language/en-GB/en-GB.com_content.ini [hunk 2]
```diff
@@ -90,6 +90,8 @@ COM_CONTENT_FIELD_PUBLISH_DOWN_DESC="An optional date to Finish Publishing the a
 COM_CONTENT_FIELD_PUBLISH_DOWN_LABEL="Finish Publishing"
 COM_CONTENT_FIELD_PUBLISH_UP_DESC="An optional date to Start Publishing the article."
 COM_CONTENT_FIELD_PUBLISH_UP_LABEL="Start Publishing"
+COM_CONTENT_FIELD_SAVEHITS_DESC="If you want to display the hits, then select YES to save the number of hits."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_content.ini]: its not display its record so

==>
"Record the number of hits."

**Review by brianteeman — CHANGES_REQUESTED** [review]

**Comment by effrit** [comment]
Cool, thanks! Tested successfully. What need to do to approve testing?

File: administrator/language/en-GB/en-GB.com_content.ini (hunk 1 above)
Comment (ggppdk) [reply administrator/language/en-GB/en-GB.com_content.ini]: Is "Count hits" a better term ?
as this feature does not keep "records" of the hits ?

**Comment by jurihahn** [comment]
@effrit go to Joomla Issue Tracker and click "Test this"
https://issues.joomla.org/tracker/joomla-cms/22413

**Comment by ghost** [comment]
@effrit if needed:

- open [Issue Tracker](https://issues.joomla.org/tracker/joomla-cms/22413)
- Login with your github-Account
- Click on blue "Test this"-Button above Authors-Picture
- mark your Test as successfully
- hit "submit test result"

**Comment by effrit** [comment]
I have tested this item :white_check_mark: successfully on ea5a30869ee9686c35a96629725c64c7669e5aa0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22413">issues.joomla.org/tracker/joomla-cms/22413</a>.</sub>

**Comment by SharkyKZ** [comment]
If accepted, shouldn't this be done for all items using hits functionality?

**Comment by jurihahn** [comment]
@SharkyKZ i can do this in future PR's

**Comment by effrit** [comment]
@SharkyKZ, which one? Examples?

**Comment by SharkyKZ** [comment]
Contacts, newsfeeds and tags. Categories also have code for hit functionality but it doesn't seem to get executed. Was thinking that, if this is added to categories, the strings should differentiate between content types. But if not, they should be made global.

Also realized hitcounts can be easily controlled by a plugin. This is already done with vote plugin to prevent increasing count when voting. That said, does an option really needs to be added? Plugin could give more control such as suggested in the forum https://forum.joomla.org/viewtopic.php?f=706&t=965788.

**Comment by effrit** [comment]
@SharkyKZ, it must be plugin AFTER this implementation done. So u will can use this BD field without standard mad increment IF u need accurate hits counter. 
And read about others request - about DB cache. With this implementations they also will be satisfied. Because if you have thousand  of articles standard hits is just pain for nothing...
So this modifications is simple and useful solution for 2 issues.

**Comment by SharkyKZ** [comment]
@effrit No, the plugin can easily disable hitcount by setting `hitcount` var to 0:

https://github.com/joomla/joomla-cms/blob/c31ff5105085ce6d3bde3ea7dea96dd538bc6fd6/components/com_content/models/article.php#L241-L256

This allows manipulation requested in the forum thread. E.g. to prevent same user from increasing the count twice or to disable robot hits.

**Comment by effrit** [comment]
@SharkyKZ, even so, why i need install the plugin if i just dont need this counter?
i guess i am not only one who never use it.

**Comment by dmitrystas** [comment]
I have tested this item :white_check_mark: successfully on ea5a30869ee9686c35a96629725c64c7669e5aa0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22413">issues.joomla.org/tracker/joomla-cms/22413</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

File: components/com_content/controller.php [hunk 3]
```diff
@@ -106,7 +108,11 @@ public function display($cachable = false, $urlparams = false)
 			// Get/Create the model
 			if ($model = $this->getModel($vName))
 			{
-				$model->hit();
+				$config   = ComponentHelper::getParams('com_content');
```
Comment (Quy) [inline components/com_content/controller.php]: Lines 111-112 can be combined.

`if (ComponentHelper::getParams('com_content')->get('save_hits', 1) == 1)`

File: administrator/components/com_content/config.xml [hunk 4]
```diff
@@ -350,6 +350,18 @@
 			<option value="0">JHIDE</option>
 		</field>
 
+		<field
+			name="save_hits"
+			type="radio"
+			label="COM_CONTENT_FIELD_SAVEHITS_LABEL"
```
Comment (Quy) [inline administrator/components/com_content/config.xml]: Change `save_hits` to `record_hits`?
Change `SAVE` to `RECORD` in the language keys?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 19dfbb3a03e9dc1db3116925e4a8bd3b08a8b5a6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22413">issues.joomla.org/tracker/joomla-cms/22413</a>.</sub>

**Comment by effrit** [comment]
I have tested this item :white_check_mark: successfully on 19dfbb3a03e9dc1db3116925e4a8bd3b08a8b5a6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22413">issues.joomla.org/tracker/joomla-cms/22413</a>.</sub>

**Comment by dmitrystas** [comment]
I have tested this item :white_check_mark: successfully on 19dfbb3a03e9dc1db3116925e4a8bd3b08a8b5a6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22413">issues.joomla.org/tracker/joomla-cms/22413</a>.</sub>

**Comment by effrit** [comment]
Hey, what else we need to do?
Who must implement changes requested by brianteeman?

**Comment by infograf768** [comment]
The PR is RTC. The changes were done. Nothing else to do than wait for @mbabker to merge. :)

File: components/com_content/controller.php [hunk 5]
```diff
@@ -106,7 +108,10 @@ public function display($cachable = false, $urlparams = false)
 			// Get/Create the model
 			if ($model = $this->getModel($vName))
 			{
-				$model->hit();
+				if (ComponentHelper::getParams('com_content')->get('record_hits', 1) == 1)
```
Comment (alikon) [inline components/com_content/controller.php:111]: ===
shouldn't be better ?

Comment (jurihahn) [reply components/com_content/controller.php:111]: Why? Type is not important.

Comment (alikon) [reply components/com_content/controller.php:111]: standard maybe ?

**Comment by SharkyKZ** [comment]
Last commit (69ea8a2205f21e2a13461ef790b528decdc0be0d) doesn't work right. If you want to use type-safe comparison, add filter to form field or at least cast value as integer.

Also, since string remains generic can you move it to global language files?

**Comment by infograf768** [comment]
Set back to pending until tested new code<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22413">issues.joomla.org/tracker/joomla-cms/22413</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 22c8e61b443509e6621d13165768908c101497aa<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22413">issues.joomla.org/tracker/joomla-cms/22413</a>.</sub>

**Comment by dmitrystas** [comment]
I have tested this item :white_check_mark: successfully on 22c8e61b443509e6621d13165768908c101497aa<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22413">issues.joomla.org/tracker/joomla-cms/22413</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22413">issues.joomla.org/tracker/joomla-cms/22413</a>.</sub>

**Pull request merged**